pkg/detectors/aws/aws.go

```diff
@@ -245,6 +266,15 @@ func awsCustomCleanResults(results []detectors.Result) []detectors.Result {
 	return out
 }
 
+type awsError struct {
+	Code    string
+	Message string
+}
+
+type awsErrorResponseBody struct {
+	Error awsError
+}
```

Comment:
**Nit:** Explicitly adding the `json` tags would help signal that this struct is used for encoding / decoding JSON. It'll also decouple the struct member names from that process.

```go
type awsError struct {
    Code    string `json:"Code"`
    Message string `json:"Message"`
}

type awsErrorResponseBody struct {
    Error awsError `json:"Error"`
}
```